### PR TITLE
add OnLoadPointListener interface

### PR DESCRIPTION
**Pull request by roideuniverse**
Ref #18 
- add `OnLoadPointListener`, for unlimited scrolling.

### CODE REVIEW

File: recycler-core/src/main/java/com/carrotcreative/recyclercore/widget/ProgressRecyclerViewLayout.java [hunk 1]
```diff
@@ -94,6 +127,51 @@ private void setDefaultAdapter()
...
+    }
+
+    /**
+     * Helper method to set the distance from bottom value programmatically, if its not set in
+     * the layout file.
+     * <p>
+     * A value of 0, means the #onReachedLoadPoint is called when the last child starts becoming
+     * visible.
+     * <p>
+     * A value of 1 means the #onReachedLoadPoint is called when the secnd last child starts
```
Comment (roideuniverse) [inline recycler-core/src/main/java/com/carrotcreative/recyclercore/widget/ProgressRecyclerViewLayout.java]: typo, `secnd`

File: recycler-core/src/main/java/com/carrotcreative/recyclercore/widget/ProgressRecyclerViewLayout.java [hunk 2]
```diff
@@ -62,6 +81,20 @@ public ProgressRecyclerViewLayout(Context context, AttributeSet attrs, int defSt
...
+        setScrollListener();
+    }
+
+    private void initAttributes(AttributeSet attrs)
+    {
+        if(attrs != null)
+        {
+            TypedArray customAttr = getContext().obtainStyledAttributes(attrs, R.styleable
+                    .ProgressRecyclerViewLayout);
+            int distanceFromButtonToRefresh = customAttr.getInt(
```
Comment (BrandonRomano) [inline recycler-core/src/main/java/com/carrotcreative/recyclercore/widget/ProgressRecyclerViewLayout.java]: `distanceFromButtonToRefresh`? Do you mean `distanceFromBottomToRefresh`?

Comment (roideuniverse) [inline recycler-core/src/main/java/com/carrotcreative/recyclercore/widget/ProgressRecyclerViewLayout.java]: 😢

File: recycler-core/src/main/java/com/carrotcreative/recyclercore/widget/UnlimitedScrollHelper.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            }
+        }
+    }
+
+    private static int getMax(int[] arr)
+    {
+        int max = arr[0];
+        for(int i = 0; i < arr.length - 1; i++)
+        {
+            max = Math.max(arr[i], arr[i + 1]);
```
Comment (BrandonRomano) [inline recycler-core/src/main/java/com/carrotcreative/recyclercore/widget/UnlimitedScrollHelper.java]: ☝️

File: recycler-core/src/main/java/com/carrotcreative/recyclercore/widget/UnlimitedScrollHelper.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        }
+        return max;
+    }
+
+    /**
+     *
+     * @param totalCount Total count of items.
+     * @param spanCount
+     * @return
```
Comment (BrandonRomano) [inline recycler-core/src/main/java/com/carrotcreative/recyclercore/widget/UnlimitedScrollHelper.java:99]: Docs could be better here ☝️

**Comment by BrandonRomano** [comment]
Back to you @roideuniverse -- primarily just that bug with `getMax`, but this looks good!

**Comment by BrandonRomano** [comment]
@roideuniverse 👍

**Pull request merged**